### PR TITLE
TemplatedType

### PR DESCRIPTION
**Pull request by varunagrawal**
This PR adds a new `TemplatedType` which lets us pass in and return templated data types, e.g. `std::vector<Link*>`.

### CODE REVIEW

File: gtwrap/interface_parser/type.py [hunk 1]
```diff
@@ -71,6 +67,7 @@ def __init__(self,
     @staticmethod
     def from_parse_result(parse_result: Union[str, list]):
         """Unpack the parsed result to get the Typename instance."""
+        print("Typename from_parse_result", parse_result, type(parse_result))
```
Comment (ProfFan) [inline gtwrap/interface_parser/type.py:70]: Do we need a print here?

File: gtwrap/interface_parser/type.py [hunk 2]
```diff
@@ -108,72 +109,76 @@ def __ne__(self, other) -> bool:
         return not res
 
 
-class QualifiedType:
-    """Type with qualifiers, such as `const`."""
+class BasicType:
+    """
+    Basic types are the fundamentla built-in types in C++ such as double, int, char, etc.
```
Comment (ProfFan) [inline gtwrap/interface_parser/type.py:114]: typo

**Review by ProfFan — APPROVED** [review]
Awesome!

File: gtwrap/interface_parser/type.py (hunk 1 above)
Comment (varunagrawal) [reply gtwrap/interface_parser/type.py:70]: Nope. Let me remove it.

File: gtwrap/interface_parser/type.py (hunk 2 above)
Comment (varunagrawal) [reply gtwrap/interface_parser/type.py]: Nice catch!

**Pull request merged**